Update Readme file (#1)
DevExpress-Examples/create-wpf-slide-view: Readme.md

[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
[![](https://img.shields.io/badge/💬_Leave_Feedback-feecdd?style=flat-square)](#does-this-example-address-your-development-requirementsobjectives)
<!-- default badges end -->

# WPF SlideView – Display a Horizontal List of Cards with Swipe Navigation

This example uses the [`SlideView`](https://docs.devexpress.com/WPF/DevExpress.Xpf.WindowsUI.SlideView) control to display a scrollable horizontal list of items. Each item is rendered as a content card with user-defined templates.

![Group Bar Items with Separators](./Images/slide-view.jpg)

Use the [`SlideView`](https://docs.devexpress.com/WPF/DevExpress.Xpf.WindowsUI.SlideView) control when you need to:

> - Present detailed information for each data item in a focused, one-at-a-time format.

Reviewer: ```suggestion
- Display detailed information for data items in a focused, single-item view.
```

> - Create a modern, user friendly interface with swipe-style navigation.

Reviewer: ```suggestion
- Create a modern, user-friendly interface with swipe-style navigation.
```

## Implementation Details

### Data Structure

> The data source contains a list of `Employee` objects, which is exposed by the `EmployeesData` view model through the `DataSource` property. The `Employee` class exposes employee-related properties (such as `FullName`, `Photo`, `JobTitle`, `EmailAddress`, and `City`). Images are stored as byte arrays and converted to BitmapImage objects at runtime.

Reviewer: ``The data source contains a list of `Employee` objects, which is exposed by the `EmployeesData` view model through the `DataSource` property.``

For my understanding, what is exposed by this view model, the data source itself or the `Employee` objects?

Author: The view model (`EmployeesData`) exposes the data source itself — a list of `Employee` objects — through the `DataSource` property.

Reviewer: I see now. In that case I think we should update this to:

```suggestion
The data source, which contains a list of `Employee` objects, is exposed by the `EmployeesData` view model through the `DataSource` property. The `Employee` class exposes employee-related properties (such as `FullName`, `Photo`, `JobTitle`, `EmailAddress`, and `City`). Images are stored as byte arrays and converted to BitmapImage objects at runtime.
```

```csharp
public class Employee {
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string FullName => FirstName + " " + LastName;
    public string JobTitle { get; set; }
    public string EmailAddress { get; set; }
    public string City { get; set; }
    public byte[] ImageData { get; set; }

    [XmlIgnore]
    public BitmapImage Photo => imageSource ??= LoadImage();

    BitmapImage LoadImage() {
        var stream = new MemoryStream(ImageData);
        var image = new BitmapImage();
        image.BeginInit();
        image.StreamSource = stream;
        image.EndInit();
        return image;
    }
}
```

### Templates

> The example defines two templates:

Reviewer: ```suggestion
The following code example defines two templates:
```

* [`ItemHeaderTemplate`](https://docs.devexpress.com/WPF/DevExpress.Xpf.WindowsUI.SlideView.ItemHeaderTemplate) displays the employee’s first name.
* [`ItemContentTemplate`](https://docs.devexpress.com/WPF/DevExpress.Xpf.WindowsUI.SlideView.ItemHeaderTemplate) displays detailed information, including photo, contact details, and job data.

```xaml
<DataTemplate x:Key="ItemHeaderTemplate">
    <TextBlock Text="{Binding FirstName}" />
</DataTemplate>

<DataTemplate x:Key="ItemContentTemplate">
    <Grid>
        <!-- Photo, FullName, JobTitle, etc. -->
    </Grid>
</DataTemplate>
```

### Slide View Configuration

```xaml
<dxwui:SlideView
    Header="Slide View"
    ItemsSource="{Binding DataSource}"
    ItemHeaderTemplate="{StaticResource ItemHeaderTemplate}"
    ItemTemplate="{StaticResource ItemContentTemplate}" />
```

## Files to Review

* [DataStorage.cs](./CS/SlideViewSample/DataStorage.cs)
* [MainWindow.xaml](./CS/SlideViewSample/MainWindow.xaml)

## Documentation

> * [SlideView](https://docs.devexpress.com/WPF/DevExpress.Xpf.WindowsUI.SlideView)

Reviewer: ```suggestion
* [SlideView Class](https://docs.devexpress.com/WPF/DevExpress.Xpf.WindowsUI.SlideView)
```

> * [SlideViewItem](https://docs.devexpress.com/WPF/DevExpress.Xpf.WindowsUI.SlideViewItem)

Reviewer: ```suggestion
* [SlideViewItem Class](https://docs.devexpress.com/WPF/DevExpress.Xpf.WindowsUI.SlideViewItem)
```

> * [ItemHeaderTemplate](https://docs.devexpress.com/WPF/DevExpress.Xpf.WindowsUI.SlideView.ItemHeaderTemplate)

Reviewer: ```suggestion
* [SlideView.ItemHeaderTemplate Property](https://docs.devexpress.com/WPF/DevExpress.Xpf.WindowsUI.SlideView.ItemHeaderTemplate)
```

> * [ItemContentTemplate](https://docs.devexpress.com/WPF/DevExpress.Xpf.WindowsUI.SlideView.ItemHeaderTemplate)

Reviewer: I think the URL is incorrect here. Can you please check?
Also, please use the full name of the help topic, like I did in my suggestions above.

Author: Ok, thanks! I will change it

Reviewer: Thanks!

* [Slide View](https://docs.devexpress.com/WPF/15020/controls-and-libraries/windows-modern-ui/content-containers/slide-view)

## More Examples

* [WPF Accordion Control – Bind to Hierarchical Data Structure](https://github.com/DevExpress-Examples/wpf-accordion-bind-to-hierarchical-data-structure)
* [WPF MVVM Framework – Use View Models Generated at Compile Time](https://github.com/DevExpress-Examples/wpf-mvvm-framework-view-model-generator)

> * [WPF Dock Layout Manager – Display ViewModels Collection in LayoutGroup](https://github.com/DevExpress-Examples/wpf-docklayoutmanager-display-viewmodels-collection-in-layoutgroup)

Reviewer: We should use the actual GitHub page title:

```suggestion
* [WPF Dock Layout Manager - Populate a DockLayoutManager LayoutGroup with the ViewModels Collection](https://github.com/DevExpress-Examples/wpf-docklayoutmanager-display-viewmodels-collection-in-layoutgroup)
```
